### DISCUSSION TITLE
Screen pages count

### DISCUSSION THREAD

**Discussion by marcelo-maalves** [readium/kotlin-toolkit#423] [Q&A]
Hi,
  I would like to know if have some solution from Readium or some sample code that I can have Screen Page counter. I need that because our application has some features that change de book screen: Font Size, Landscape to Portrait, etc. 

Thanks! 
   Marcelo Alves

**Comment by mickael-menu** [comment]
There's no way to compute the screen pages for the whole publication, as we don't do background pre-rendering.

However you can get the screen pages count for the current resource by providing a `PaginationListener` when creating the `EpubNavigatorFragment`: https://github.com/readium/kotlin-toolkit/blob/011e0d74adc66ec2073f746d815310b838af4fbf/readium/navigator/src/main/java/org/readium/r2/navigator/epub/EpubNavigatorFragment.kt#L1060

**Reply by tanel-oun** [reply]
Hello, @mickael-menu 
I am reaching out from project management side for this application problem and I like to know if it would be possible and you would be interested building this feature for toolkit.
If yes, then we can discuss about specific points via e-mail.

**Reply by mickael-menu** [reply]
Hi, computing screen pages for the whole publication is not something we're interested in for Readium at the moment. Doing it would require significant changes in the architecture of the navigator to be able to render resources in the background.

**Reply by tanel-oun** [reply]
Hello, 
Thank you for your reply. I would like to add, this would be paid job, if the estimation is agreeable by different sides. Again, specific points could be discussed via email.
